planning/behavior_velocity_planner/src/utilization/util.cpp

```diff
@@ -244,6 +244,78 @@ double calcJudgeLineDistWithJerkLimit(
   return std::max(0.0, x1 + x2 + x3);
 }
 
+double findReachTime(
+  const double j, const double a, const double v, const double d, const double min,
+  const double max)
+{
+  auto f = [](const double t, const double j, const double a, const double v, const double d) {
+    return j * t * t * t / 6.0 + a * t * t / 2.0 + v * t - d;
+  };
+  const double eps = 1e-5;
+  const int max_iter = 100;
+  double lower = min;
+  double upper = max;
+  double t;
+  int iter = 0;
+  for (int i = 0; i < max_iter; i++) {
+    t = 0.5 * (lower + upper);
+    const double fx = f(t, j, a, v, d);
+    // std::cout<<"fx: "<<fx<<" up: "<<upper<<" lo: "<<lower<<" t: "<<t<<std::endl;
+    if (std::abs(fx) < eps) {
+      break;
+    } else if (fx > 0.0) {
+      upper = t;
+    } else {
+      lower = t;
+    }
+    iter++;
+  }
+  // std::cout<<"iter: "<<iter<<std::endl;
+  return t;
+}
+
+double calculateMaxSlowDownVelocity(
+  const double max_slowdown_jerk, const double max_slowdown_accel, const double current_accel,
+  const double current_velocity, const double distance_to_target)
```

Comment:
The problem setting is stopping with acceleration and jerk limitation, right?
Could you give me a mathematical formulation of this function. I will check if the formulation is correct or not.

Reply:
@takayuki5168 
mathmatical formula is almost then same as description
and because this function considers stop distance with maximum jerk and maximum acceleration judgement whether this stop can be during const jerk or const accel is calculated from t1 
![image](https://user-images.githubusercontent.com/65527974/154009294-22014130-81ba-48bd-9247-e075cf0d2c11.png)